### PR TITLE
chore: add pre-commit and fix linter step in CI

### PR DESCRIPTION
**Pull request by omrozowicz-splunk**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+    -   id: check-merge-conflict
+-   repo: https://github.com/psf/black
+    rev: 21.7b0
```
Comment (arys-splunk) [inline .pre-commit-config.yaml:22]: This can be updated to 21.8b0

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    hooks:
+    -   id: black
+-   repo: https://github.com/PyCQA/isort
+    rev: 5.9.3
+    hooks:
+    -   id: isort
+        args: ["--profile", "black"]
```
Comment (arys-splunk) [inline .pre-commit-config.yaml:29]: Isn't this setting already set in `pyproject.toml`?

**Review by arys-splunk — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (omrozowicz-splunk) [reply .pre-commit-config.yaml:29]: It is, but when I run it locally to automatically fix files before commit it doesn't use pyproject.toml

Comment (arys-splunk) [reply .pre-commit-config.yaml:30]: Sounds strange, it seems to be working for other repositories.

Comment (arys-splunk) [reply .pre-commit-config.yaml:30]: Can you provide an example of it not taking `pyproject.toml` into account?

Comment (omrozowicz-splunk) [reply .pre-commit-config.yaml]: I can't reproduce this issue, so `args: ["--profile", "black"]` is not necessary.

**Pull request merged**